src/main/java/be/seeseemelk/mockbukkit/entity/EndermanMock.java

```diff
@@ -79,6 +84,36 @@ public void setCarriedBlock(@Nullable BlockData blockData)
 		this.carriedBlock = blockData;
 	}
 
+	@Override
+	public boolean teleport()
+	{
+		if (alive)
+		{
+			double x = this.getLocation().x() + (random.nextDouble() - 0.5D) * 64.0D;
+			double y = this.getLocation().y() + (double) (random.nextInt(64) - 32);
+			double z = this.getLocation().z() + (random.nextDouble() - 0.5D) * 64.0D;
+			return teleport(new Location(getWorld(), x, y, z));
+		}
+		return false;
+	}
+
+	@Override
+	public boolean teleportTowards(@NotNull Entity entity)
+	{
+		if (alive)
+		{
+			Vector vector = new Vector(this.getLocation().x() - entity.getLocation().x(),
+					(this.getLocation().y() + 1.45) - entity.getLocation().y(), this.getLocation().z() - entity.getLocation().z());
+
+			vector = vector.normalize();
+			double x = this.getLocation().x() + (random.nextDouble() - 0.5D) * 8.0D - vector.getX() * 16.0D;
+			double y = this.getLocation().y() + (double) (random.nextInt(16) - 8) - vector.getY() * 16.0D;
+			double z = this.getLocation().z() + (random.nextDouble() - 0.5D) * 8.0D - vector.getZ() * 16.0D;
+			return this.teleport(new Location(getWorld(), x, y, z));
+		}
+		return false;
+	}
```

Comment:
Can you also please add Tests for this? It should be possible to just check if the Location after the method call is different from the one beforehand.

Reply:
Maybe also check that the enderman is closer to the player than before

Reply:
not crazy about testing distance - as it doesn't necessarily have to be true
example: 
```java
@Test
void testRandomTeleportToEntity()
{
        Location loc = endermanMock.getLocation();
	PlayerMock player = serverMock.addPlayer();
	WorldMock mock = serverMock.addSimpleWorld("world");
	player.teleport(new Location(mock, 20, 20, 20));
	endermanMock.teleport(new Location(mock, 1, 1, 1));
	double distance = endermanMock.getLocation().distanceSquared(player.getLocation());
	assertTrue(endermanMock.teleportTowards(player));
	assertNotEquals(loc, endermanMock.getLocation());
	assertTrue(distance > endermanMock.getLocation().distanceSquared(player.getLocation()));
}
```
Distance test is not concrete- if the player is teleported to 5, 5, 5, the test fails, but at 20, 20, 20, the test passes